playbook/app/pb_kits/playbook/pb_star_rating/star_rating.html.erb

```diff
@@ -28,13 +28,33 @@
     <% end %>
   <% end %>
   <%= pb_rails("flex", props: { }) do %>
-    <% object.star_count.times do %>
-        <%= pb_rails("icon", props: { classname: "#{star_color} pb_star_#{size}" , custom_icon: Playbook::Engine.root.join(star_svg_path) } ) %>
-    <% end %>
-    <% object.empty_stars.times do %>
-        <%= pb_rails("icon", props: { classname: "#{background_star_color} pb_star_#{size}", custom_icon: Playbook::Engine.root.join(background_star_path) } ) %>
+
+    <% if object.variant == "display" %>
+
+      <% object.star_count.times do %>
+          <%= pb_rails("icon", props: { classname: "#{star_color} pb_star_#{size}" , custom_icon: Playbook::Engine.root.join(star_svg_path) } ) %>
+      <% end %>
+      <% object.empty_stars.times do %>
+          <%= pb_rails("icon", props: { classname: "#{background_star_color} pb_star_#{size}", custom_icon: Playbook::Engine.root.join(background_star_path) } ) %>
+      <% end %>
+  
+    <% else %>
+      <%= pb_rails("flex", props: { orientation: "column" }) do %>
+        <% if object.label.present? %>
+          <%= pb_rails("caption", props: {text: object.label, margin_bottom:"xs"}) %>
+        <% end %>
+        <input type="hidden" id="star-rating-input" value="" name="<%= object.name %>"/>
+        <%= pb_rails("flex", props: { orientation: "row" }) do %>
+          <% object.denominator.times do |index| %>
+            <div data-pb-star-rating id="<%= index + 1 %>" class="<%= star_color %>">
+              <%= pb_rails("icon", props: { classname: "#{background_star_color} pb_star_#{size} interactive-star-icon", custom_icon: Playbook::Engine.root.join(background_star_path)} ) %>
+            </div>
+          <% end %>
+        <% end %>
+      <% end %>
```

Comment:
Just a thought but these could be sub components or maybe even a function to make it a bit easier to read.

Reply:
Hey! i tried breaking this up into partials but am running into issues where the partial files cant be found. Since there is more work to be done on this kit, we can look into a better way to this on the follow up stories. thanks!!